backend/src/main/java/gov/cdc/usds/simplereport/api/apiuser/UserAccountCreationController.java

```diff
@@ -133,9 +132,52 @@ public void enrollEmailMfa(@RequestBody EnrollMfaRequest requestBody, HttpServle
    * @param request contains session information about the user, including their Okta id.
    * @throws OktaAuthenticationFailureException if Okta cannot enroll the user in MFA.
    */
-  @GetMapping("/authenticator-qr")
-  public void getAuthQrCode(@RequestBody EnrollMfaRequest requestBody, HttpServletRequest request) {
-    // WIP: doesn't interact with Okta yet.
+  @PostMapping("/authenticator-qr")
+  public FactorAndQrCode getAuthQrCode(
+      @RequestBody EnrollMfaRequest requestBody, HttpServletRequest request)
+      throws OktaAuthenticationFailureException {
+    String userId = getUserId(request.getSession());
+    FactorAndQrCode factorData =
+        _oktaAuth.enrollAuthenticatorAppMfa(userId, requestBody.getUserInput());
+    request.getSession().setAttribute(FACTOR_ID_KEY, factorData.getFactorId());
+    return factorData;
+  }
+
+  /**
+   * Enrolls a security key for the user, returning data needed to finish activation.
+   *
+   * @param request contains session information about the user, including their Okta id.
+   * @return activation object with information to finish enrolling the security key, including a
+   *     challenge and user id.
+   * @throws OktaAuthenticationFailureException if the user is not recognized or Okta cannot enroll
+   *     their security key.
+   */
+  @PostMapping("/enroll-security-key-mfa")
+  public String enrollSecurityKeyMfa(HttpServletRequest request)
+      throws OktaAuthenticationFailureException {
+    String userId = getUserId(request.getSession());
+    JSONObject enrollResponse = _oktaAuth.enrollSecurityKey(userId);
+    request.getSession().setAttribute(FACTOR_ID_KEY, enrollResponse.getString(FACTOR_ID_KEY));
+    return new JSONObject(enrollResponse, "activation").toString();
```

Comment:
I think we're going to push this off to a followup PR if that's alright, since we'd like to get this out ASAP so we can begin user testing. But I'll flag it with a TODO and follow up with a PR by the end of the week!